ansible_collections/arista/avd/roles/eos_designs/python_modules/overlay/router_bgp.py

```diff
@@ -166,6 +177,15 @@ def _peer_groups(self) -> list | None:
 
         return peer_groups
 
+    def _generate_bfd_timers(self, peer_group) -> dict:
+        bfd_timers = {}
+        if self.shared_utils.bgp_peer_groups[peer_group]["bfd"]:
+            # assigning default value here.
+            bfd_timers = DEFAULT_BFD_TIMERS_PER_PEER_GROUP.get(peer_group)
+            if "bfd_timers" in self.shared_utils.bgp_peer_groups[peer_group]:
+                bfd_timers = self.shared_utils.bgp_peer_groups[peer_group]["bfd_timers"]
+        return bfd_timers
```

Comment:
We need to handle the case of `None` as well, so we have some helper functions to do that. `get()` will walk nested dicts according to the dot-notation path in the second argument. If any key is missing or None, it will return the default value.

```suggestion
        default_bfd_timers = DEFAULT_BFD_TIMERS_PER_PEER_GROUP.get(peer_group)
        return get(self.shared_utils.bgp_peer_groups, f"{peer_group}.bfd_timers", default=default_bfd_timers)
```